### ISSUE TITLE
transformJsxProps errors on large strings

### ISSUE THREAD

**Issue by joprice** [fable-compiler/Fable#3839]
### Description

There seem to be inputs of a certain size that trigger a transformation from a string like `"a"` to the expression `let x = "a" in x`. Since [`transformJsxProps` is checking for a specific pattern of a tuple expression](https://github.com/fable-compiler/Fable/blob/e2285d6e6f914f3f622bcb3165a23eca0153b856/src/Fable.Transforms/Fable2Babel.fs#L2031) for the prop's key and value, the pattern match fails and the error `Cannot detect JSX prop key at compile time` is triggered. This can be worked around by moving the literal into a local variable, but some frontend patterns like long tailwind classes, hardcoded urls, and svg values are often longer than 100 characters, making it an inconvenient limitation.

Changing the pattern match to destructure the Let binding works, but I'm unsure there aren't cases where this would produce an incorrect result:

```fsharp
 | Some(props, children),
                (Fable.Value(Fable.NewTuple([ StringConst key; value ], _), _)
              | Fable.Let(_, value, Fable.Value(Fable.NewTuple([ StringConst key ; _], _), _))) ->
```

### Repro code

[Please provide the F# code to reproduce the problem or a link to the [REPL](https://fable.io/repl/).
Ideally, it should be possible to easily turn this code into a unit test.](https://github.com/fable-compiler/Fable/compare/main...joprice:Fable:jsxError)

### Expected and actual results

Literals of any size should not cause compilation errors or require workarounds to use in JSX expressions.

### Related information

* Fable version: `4.18.0`
* Operating system: OSX

**Comment by joprice** [comment]
If I increase the limit from 100 to 200, it does work as expected, so I think it's a case like the following

With an inline definition like

```fsharp 
static member inline src (value: string): JSX.Prop = "src", value
```

and an argument to the inline function that is considered small enough:

```fsharp
Attr.src "a"
// becomes
("src", "a")
```

When the argument to the inline function is considered too large:

```fsharp
Attr.src "a......"
// becomes
let x = "a......"
("src", x)
```

However, it seems that this is only useful when the input is a let binding with more than one reference, not when it's a literal.

```fsharp
let veryLongStringThatCouldBeInlined = "a......."
("src", veryLongStringThatCouldBeInlined)
```


```fsharp
let veryLongStringThatShouldntBeInlined = "a......."
("src", veryLongString)
("src", veryLongString)
```

But maybe there's a case for it I'm not aware of.
